File: ed1/gara3/parentesi/att/parentesi.c
#include <stdio.h>
#include <assert.h>

#define MAXN 10000

int controlla(int N, char E[]) {
    // Mettete qui il codice della soluzione
    return -1;
}


char E[MAXN + 5]; // Maggiore di MAXN per contenere newline e null

int main() {
    FILE *fr, *fw;
    int N, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(1 == fscanf(fr, "%d\n", &N));
    assert(fgets(E, MAXN + 5, fr) == E);
    E[N] = 0;
    
    if (controlla(N, E) < 0)
        fprintf(fw, "malformata\n");
    else
        fprintf(fw, "corretta\n");
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed1/gara3/parentesi/att/parentesi.pas
function controlla(N: longint; var E: array of char): longint;
begin
    (* Mettete qui il codice della soluzione *)
    controlla := 0;
end;


const
    MAXN = 10000;
var
    N     : longint;
    E     : array[0..MAXN] of char;
    fr, fw: text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N);
    readln(fr, E);

    if (controlla(N, E) < 0) then
        writeln(fw, 'malformata')
    else
        writeln(fw, 'corretta');
    close(fr);
    close(fw);
end.
